go/client/chat_cli_rendering.go

```diff
@@ -0,0 +1,132 @@
+package client
+
+import (
+	"fmt"
+
+	humanize "github.com/dustin/go-humanize"
+	"github.com/keybase/client/go/libkb"
+	"github.com/keybase/client/go/protocol/chat1"
+	"github.com/keybase/client/go/protocol/gregor1"
+	"github.com/keybase/client/go/protocol/keybase1"
+)
+
+type conversationListView []keybase1.ConversationLocal
+
+// maxWidth must >= 3
+func (v conversationListView) renderConversationName(maxWidth int, index int) string {
+	if v[index].Info == nil {
+		return ""
+	}
+	name := v[index].Info.TlfName
+	if len(v[index].Messages) > 0 && v[index].Messages[0].Info != nil && v[index].Messages[0].Info.IsNew {
+		name = "* " + name
+	}
+	if len(name) > maxWidth {
+		name = name[:maxWidth-3] + "..."
+	}
+	return name
+}
+
+func (v conversationListView) show(ui libkb.TerminalUI) {
+	if len(v) == 0 {
+		return
+	}
+	iConversation := 0
+	ui.TablifyAlignRight(nil, func() []string {
+		for ; iConversation < len(v) && len(v[iConversation].Messages) == 0; iConversation++ {
+		}
+		if iConversation == len(v) {
+			return nil
+		}
+		ret := []string{
+			fmt.Sprintf("[%d]", iConversation),
+			v.renderConversationName(20, iConversation),
+			messageFormatter(v[iConversation].Messages[0]).renderAuthorAndTime(),
+			messageFormatter(v[iConversation].Messages[0]).renderMessage(40),
```

Comment:
where does the 40 come from?

We can get the terminal size in the `Terminal` class. I would expose it to the `TerminalUI` interface and then call it here..